### PR TITLE
Feature #2566 v11.1.0-rc1

### LINKED ISSUE #2566: Create the MET-11.1.0-rc1 release
## Describe the Task ##

Create the MET-11.1.0-rc1 development release. Remember to create a new discussion for testing the rc1 release, as we did for beta1 with dtcenter/METplus#2039 and beta2 with dtcenter/METplus#2135.

### Time Estimate ###
2 hours

### Sub-Issues ###
Consider breaking the task down into sub-issues.
None needed.

### Relevant Deadlines ###
Should be created on Monday 6/12/23 or Tuesday 6/13/23.

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
No impacts.

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by JohnHalleyGotway**
## Expected Differences ##

Adds release notes for the 11.1.0-rc1 release. Also adds logic to replace '/' with '_' in the Docker tag names used by GHA. This logic was needed for the recent PR #2556 from a fork with a branch named `feature/compile_mac_clang14.x`. The '/' character is not allowed in Docker tag names.

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **[No]**</br>
If **yes**, please describe:</br>

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Confirmed that GHA successfully compiles with the code even with the script changes.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Please review release notes [here](https://met.readthedocs.io/en/feature_2566_v11.1.0-rc1/Users_Guide/release-notes.html#met-version-11-1-0-rc1-release-notes-20230613)

I listed the date as tomorrow (20230614) since it's not getting done today.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[20230614]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: .github/jobs/bash_functions.sh [hunk 1]
```diff
@@ -25,3 +25,13 @@ function time_command {
   fi
   return $error
 }
+
+# utility function to construct the DockerHub tag name to be used,
+# replacing slashes with underscores
+
+function get_dockerhub_tag {
+
+  tag_name=`echo ${DOCKERHUB_REPO}:${SOURCE_BRANCH} | sed 's%/%_%g'`
```
Comment (georgemccabe) [inline .github/jobs/bash_functions.sh]:
```suggestion
  echo ${DOCKERHUB_REPO}:${SOURCE_BRANCH} | sed 's%/%_%g';
```
You can't return a string in a bash function, only a return value integer, so just echo the value. This is output when I run this in a shell:

> -bash: return: dtcenter_met:my_branch_name: numeric argument required

File: .github/jobs/bash_functions.sh [hunk 2]
```diff
@@ -25,3 +25,13 @@ function time_command {
...
+
+  return $tag_name;
```
Comment (georgemccabe) [inline .github/jobs/bash_functions.sh]:
```suggestion
```
Remove this return

File: .github/jobs/build_docker_image.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 source ${GITHUB_WORKSPACE}/.github/jobs/bash_functions.sh
 
-DOCKERHUB_TAG=${DOCKERHUB_REPO}:${SOURCE_BRANCH}
+DOCKERHUB_TAG=get_dockerhub_tag
```
Comment (georgemccabe) [inline .github/jobs/build_docker_image.sh]:
```suggestion
DOCKERHUB_TAG=`get_dockerhub_tag`
```
With the above changes, the function will echo the value and this change will set the echo'd value to the DOCKERHUB_TAG variable.

**Review by georgemccabe — CHANGES_REQUESTED** [review]
The compilation step fails because it is setting the DockerHub tag to the string `get_dockerhub_tag`. 

> The push refers to repository [docker.io/library/get_dockerhub_tag]

I made suggested changes that I think should fix it. I tested them in a bash shell locally.

**Comment by JohnHalleyGotway** [comment]
Thanks for catching these issues. Sorry, don’t know how I missed this. I could’ve sworn that it complied fine via GHA but perhaps I just saw the successful documentation workflow run? I’ll get this patched up tomorrow and merge after a successful compilation.

**Comment by JohnHalleyGotway** [comment]
Re-requesting review. The [GHA run](https://github.com/dtcenter/MET/actions/runs/5269432626/jobs/9528400206?pr=2573) compiled fine. It does flag diffs but those are related to recently approved PR's and not this one.

**Review by georgemccabe — APPROVED** [review]
The compilation step is now working properly.
I reviewed the release notes and they look good. All items are descriptive and clear. The links are all correct as well.
I approve.

**Pull request merged**